src/formatDate.js

```diff
@@ -9,6 +9,58 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateSeparated = date.split(fromFormat[3]);
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < 3; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+      case 'YY':
+        year = dateSeparated[i];
+        break;
+      case 'MM':
+        month = dateSeparated[i];
+        break;
+      case 'DD':
+        day = dateSeparated[i];
+        break;
+    }
```

Comment:
There is missing default case where you could throw an error or smth that the format is unprocessable.